src/MenuPreprocess.php

```diff
@@ -0,0 +1,208 @@
+      array_shift($links);
+    }
+
+    $variables['button'] = $button;
+    $this->addDropbuttonClasses($links);
+  }
+
+  /**
+   * Implements hook_preprocess_menu_local_tasks().
+   *
+   * Preprocess local tasks as BCL menu links.
+   *
+   * @param array $variables
+   *   The variables array.
+   */
+  public function preprocessMenuLocalTasks(array &$variables): void {
+    // Sort local tasks by weight.
+    uasort($variables['primary'], [
+      '\Drupal\Component\Utility\SortArray',
+      'sortByWeightProperty',
+    ]);
+    uasort($variables['secondary'], [
+      '\Drupal\Component\Utility\SortArray',
+      'sortByWeightProperty',
+    ]);
+
+    $variables['primary'] = $this->prepareLocalTasks($variables['primary']);
+    $variables['secondary'] = $this->prepareLocalTasks($variables['secondary']);
+  }
+
+  /**
+   * Adds required classes to dropbutton links.
+   *
+   * @param array $links
+   *   Array with links items.
+   */
+  public function addDropbuttonClasses(array &$links): void {
+    foreach ($links as $key => $link) {
+      if (isset($links[$key]['link']) && is_array($links[$key]['link'])) {
+        $links[$key]['link']['#options']['attributes']['class'][] = 'dropdown-item';
+      }
+      if (isset($links[$key]['text_attributes'])) {
+        $links[$key]['text_attributes']->addClass('dropdown-item');
+      }
+      if (isset($links[$key]['attributes'])) {
+        $links[$key]['attributes']->addClass('dropdown-item');
+      }
+      if (isset($links[$key]['text']) && is_array($links[$key]['text'])) {
+        $links[$key]['text']['#attributes']['class'][] = 'dropdown-item';
+      }
+    }
+  }
+
+  /**
+   * Adapts a menu link to BCL requirements.
+   *
+   * @param array $menu_link
+   *   The menu link to alter.
+   * @param array $extra_classes
+   *   Extra classes for the link.
+   *
+   * @return array
+   *   The altered menu link.
+   */
+  public function menuLink(array $menu_link, array $extra_classes = []): array {
+    $link = $menu_link + [
+      'label' => $menu_link['title'],
+      'path' => $menu_link['url'],
+    ];
+
+    $attributes = [];
+    if (!empty($menu_link['attributes'])) {
+      $attributes = $menu_link['attributes'] instanceof Attribute ? $menu_link['attributes']->toArray() : $menu_link['attributes'];
+    }
+    $class = $attributes['class'] ?? [];
+    $class = array_merge($class, $extra_classes);
+
+    if (!empty($menu_link['in_active_trail'])) {
+      $class[] = 'active';
+    }
+
+    $attributes['class'] = $class;
+    $link['attributes'] = new Attribute($attributes);
+
+    return $link;
+  }
+
+  /**
+   * Prepare a list of local task menu links.
+   *
+   * @param array $local_tasks
+   *   The local tasks.
+   *
+   * @return array
+   *   A list of filtered and processed links.
+   */
+  public function prepareLocalTasks(array $local_tasks): array {
+    $links = [];
+    foreach ($local_tasks as $link) {
+      if ($link['#access']->isForbidden()) {
```

Comment:
Core handles this differently: see build/core/lib/Drupal/Core/Render/Renderer.php:226.
Basically access is granted only if access is set to allowed, or to true.
In our case we will always have objects (\Drupal\Core\Menu\LocalTaskManager::getTasksBuild()).